test/open_api/decode_test.exs

```diff
@@ -0,0 +1,385 @@
+                 description: "Find more info here",
+                 url: "https://example.com"
+               },
+               name: "pet"
+             } == tag
+
+      assert [
+               %{
+                 "petstore_auth" => ["write:pets", "read:pets"]
+               }
+             ] == security
+
+      assert %{
+               "/example" => %OpenApiSpex.PathItem{
+                 summary: "/example summary",
+                 description: "/example description",
+                 servers: [%OpenApiSpex.Server{}],
+                 parameters: [
+                   %OpenApiSpex.Reference{
+                     "$ref": "#/components/parameters/ContentTypeHeader"
+                   }
+                 ],
+                 post: %OpenApiSpex.Operation{
+                   parameters: [
+                     %OpenApiSpex.Reference{},
+                     %OpenApiSpex.Reference{},
+                     %OpenApiSpex.Parameter{}
+                   ],
+                   deprecated: false,
+                   operationId: "example-post-test",
+                   requestBody: requestBody,
+                   callbacks: operationCallbacks,
+                   responses: operationResponses,
+                   security: operationSecurity,
+                   tags: ["test"],
+                   summary: "/example post summary",
+                   description: "/example post description",
+                   externalDocs: %OpenApiSpex.ExternalDocumentation{
+                     description: "Find more info here",
+                     url: "https://example.com"
+                   }
+                 }
+               }
+             } = paths
+
+      assert [
+               %{
+                 "petstore_auth" => ["write:pets"]
+               }
+             ] == operationSecurity
+
+      assert %{
+               "200" => %OpenApiSpex.Response{}
+             } = operationResponses
+
+      assert %{
+               "operationCallback" => %{
+                 "http://server-a.com?transactionId={$request.body#/id}" =>
+                   %OpenApiSpex.PathItem{}
+               }
+             } = operationCallbacks
+
+      assert %OpenApiSpex.Schema{
+               properties: properties
+             } =
+               get_in(
+                 requestBody,
+                 [:content, "application/json", :schema]
+                 |> Enum.map(&Access.key/1)
+               )
+
+      passengers =
+        get_in(
+          properties,
+          [:data, :properties, :passengers]
+          |> Enum.map(&Access.key/1)
+        )
+
+      assert %OpenApiSpex.Schema{} = passengers
+
+      assert %OpenApiSpex.Schema{
+               type: :string,
+               enum: ["adult", "child"]
+             } =
+               get_in(
+                 passengers,
+                 [:items, :properties, :type]
+                 |> Enum.map(&Access.key/1)
+               )
+
+      test_conn =
```

Comment:
This feels like the start of a new test.

Reply:
Agreed, happy to address this!